### PR TITLE
libnetwork/drivers/overlay: only program xt_bpf rules

### PR DESCRIPTION
**Pull request by corhere**
Drop support for platforms which only have `xt_u32` but not `xt_bpf`. No attempt is made to clean up old `xt_u32` iptables rules left over from a previous daemon instance.

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

- The `xt_bpf` kernel module is now required to use encrypted overlay networks

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: libnetwork/drivers/overlay/bpf.go [hunk 1]
```diff
@@ -45,3 +46,15 @@ func marshalXTBPF(prog []bpf.RawInstruction) string { //nolint:unused
 	}
 	return b.String()
 }
+
+// matchVXLANWithBPF returns an iptables rule fragment which matches VXLAN
```
Comment (neersighted) [inline libnetwork/drivers/overlay/bpf.go:50]:
```suggestion
// matchVXLAN returns an iptables rule fragment which matches VXLAN
```

**Review by neersighted — APPROVED** [review]
LGTM, with one comment nit.

**Comment by akerouanton** [comment]
> Drop support for platforms which only have xt_u32 but not xt_bpf

Is there any Linux distribution that don't ship xt_bpf yet?

**Comment by neersighted** [comment]
It's present in RHEL 7.9's 3.10; since it's a default netfilter extension, it's more a matter of "are there platforms that deliberately exclude this module, and dow e care about them?"

It would be good to get some more confirmation of availability before merge, across the platforms that Docker CE and Mirantis Container Runtime support.

**Comment by thaJeztah** [comment]
> It's present in RHEL 7.9's 3.10

Does that exclude RHEL 7.8 and below?

From a "docker ce" perspective, we always considered CentOS to be a "rolling release" (so effectively only "latest" 7, 8, 9 are actual releases). That said, I frequently encounter users that may have "upgraded" their install .... except for the kernel 🙈, or running older versions as a "free" alternative to match their RHEL servers.

IBM also builds binaries of Docker CE (which we publish on download.docker.com), which includes RHEL packages for s390x (CentOS does not support that architecture, but RHEL does). 

For my understanding (and if older 7.x versions do not support this) what is the effect of this patch for those platforms? (and how does it surface to the user)?

**Comment by neersighted** [comment]
We haven't confirmed below 7.9 because we do not currently support those platforms; I agree that 'latest' is the gold standard, and by that standard, we do support CentOS 7. Still, it could be interesting/reassuring to characterize presence in older CentOS/RHEL 7 kernels.

**Comment by corhere** [comment]
`xt_bpf.ko` is shipped with CentOS `kernel-3.10.0-123.el7` [(the kernel version which shipped with CentOS 7.0)](https://en.wikipedia.org/wiki/CentOS#CentOS_version_7) according to [the repo filelists.](https://vault.centos.org/7.0.1406/os/x86_64/repodata/7a90ec1358cb1d9cab4c84793f9b96602fb49e7db90a6b8f5ddc441cb2a918c6-filelists.xml.gz) 
```xml
  <file>/lib/modules/3.10.0-123.el7.x86_64/kernel/net/netfilter/xt_bpf.ko</file>
```

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks for checking kernel support!

**Pull request merged**